pandas/plotting/_core.py

```diff
@@ -861,11 +863,42 @@ def _plot_colorbar(self, ax, **kwds):
 class ScatterPlot(PlanePlot):
     _kind = 'scatter'
 
-    def __init__(self, data, x, y, s=None, c=None, **kwargs):
+    def __init__(self, data, x, y, s=None, c=None, size_factor=1, **kwargs):
         if s is None:
-            # hide the matplotlib default for size, in case we want to change
-            # the handling of this argument later
-            s = 20
+            # Set default size if no argument is given.
+            s = 20 * size_factor
+        elif is_hashable(s) and s in data.columns:
+            # Handle the case where s is a label of a column of the df.
```

Comment:
Can you separate the bubble type and size logic from the getting / setting on self. I'd like for testing that the bubble sizes are computed correctly to be easier.

So a standalone function like

```python
def get_bubbles(size_data, bubble_points):
    ...
    return bubble_dtype, s_data_max, s
```

Reply:
Thanks Tom for the review,
I separated the bubble logic into two functions, one for the bubbles in the plot, one for the bubbles in the legend. For the bubbles in the plot, the sizes are passed to the parent class as an argument `s`. For the legend, I stored the sizes and labels as attributes of the scatter plot which are used by the legend building functions. Let me know if that's what you had in mind.

Reply:
Mainly, I'm interested in ease of testing. I'd like to have a dedicated method that, given an array of values, tells me what the marker size should be for each point.